core/src/models/Destination.ts

```diff
@@ -240,6 +300,46 @@ export class Destination extends LoggedModel<Destination> {
     return DestinationOps.unTrackGroup(this);
   }
 
+  async getSupportedSyncModes() {
+    return DestinationOps.getSupportedSyncModes(this);
+  }
+
+  async getSyncMode() {
+    if (this.syncMode) return this.syncMode;
+
+    const supportedSyncModes = await this.getSupportedSyncModes();
+    if (supportedSyncModes.length === 1) {
+      // If destination only supports one sync mode, use it
+      return supportedSyncModes[0];
+    } else if (supportedSyncModes.length === 0) {
+      // If destination does not support sync modes, let it create, update, and delete
+      return "sync";
+    }
+
+    return null;
+  }
+
+  async validateSyncMode() {
+    if (this.state !== "ready") return;
```

Comment:
I'm reading that we we want to validate this as a part of becoming ready, with the `@BeforeSave` below.   I'd suggest that you remove this line so the validation method can be used more generically and then in the hook below we'll put the logic to check only if the instance is ready (or trying to become ready).